### DISCUSSION TITLE
Return associated Source model on `Target.create()`

### DISCUSSION THREAD

**Discussion by zaxovaiko** [sequelize/sequelize#15186] [Help & Questions] [answered]
## Issue Creation Checklist

[x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

It's more like a question rather than an issue. There is nothing related to it in the docs, so I decided to ask here.   I have two models `A` and `B`. 

```js
A.hasOne(B);
B.belongsTo(A);
```

When I create an instance of model `A` I want it to be returned with association `B`. Something like this:
```txt
A {
 id, name, ..., B_id, B { id, another_column, ... }
}
```

I'm using usual `create` method including model `A`:

```js
await B.create(fields, { include: A });
```

But at the response from the `sequelize` I'm getting correct and fulfilled fields of model `B`, but `null` for model `A`. What should I do to get the fields of model `A` in `B.create()`? Is it even possible? 

I found [this question](https://stackoverflow.com/questions/67837154/how-to-return-just-created-row-with-its-association-with-sequelize) and [this](https://stackoverflow.com/a/51959381/7791165) on StackOverflow, but the only answer for it is not the best, because we need to make another call to the db to fetch the created instance. So.. is there any solution for that?

I found out that I can use `B.create().then(m => m.reload())`, but still it's something that I would like to avoid if there is better option.

### Additional context

![image](https://user-images.githubusercontent.com/34129845/197834192-2ff9e542-d0f2-4eb3-ae8d-4d2ca0f63fbf.png)
![image](https://user-images.githubusercontent.com/34129845/197834343-d368475c-c9f3-405a-b3fd-f160b18ea286.png)

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [x] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment, marked as answer]
The `include` parameter in `create` is only used to insert associated entities at the same time, but it is unable to fetch it. We'd need to make a separate `SELECT` query for it so it wouldn't be more performant than if you did that yourself. I suppose we could consider doing it but it's something we'd need to analyze thoroughly first

**Reply by zaxovaiko** [reply]
Ok, thanks a lot for the answer.
